Examples/Detectors/GenericDetector/src/GenericDetector.cpp

```diff
@@ -323,6 +323,427 @@ Gen1GenericDetectorBuilder::buildTrackingGeometry(
   return trackingGeometry;
 }
 
+class Gen3GenericDetectorBuilder : public GenericDetectorBuilder {
+ public:
+  using GenericDetectorBuilder::GenericDetectorBuilder;
+
+  std::unique_ptr<const Acts::TrackingGeometry> buildTrackingGeometry(
+      const Acts::GeometryContext& gctx,
+      std::optional<std::filesystem::path> graphvizFile);
+
+  void buildPixel(Acts::Experimental::BlueprintNode& parent,
+                  const Acts::GeometryContext& gctx);
+
+  void buildShortStrip(Acts::Experimental::BlueprintNode& parent,
+                       const Acts::GeometryContext& gctx);
+
+  void buildLongStrip(Acts::Experimental::BlueprintNode& parent,
+                      const Acts::GeometryContext& gctx);
+
+  static std::shared_ptr<const Acts::HomogeneousSurfaceMaterial> asHomogeneous(
+      std::string_view debugLabel,
+      const std::shared_ptr<const Acts::ISurfaceMaterial>& material) {
+    auto hm = std::dynamic_pointer_cast<const Acts::HomogeneousSurfaceMaterial>(
+        material);
+    if (hm == nullptr) {
+      throw std::runtime_error(std::string{debugLabel} +
+                               " material is not homogeneous");
+    }
+    return hm;
+  }
+
+  using LayerType = Acts::SurfaceArrayNavigationPolicy::LayerType;
+  std::unique_ptr<Acts::NavigationPolicyFactory> createNavigationPolicy(
+      LayerType layerType,
+      std::pair<std::size_t, std::size_t> bins = {0, 0}) const {
+    using SrfArrayNavPol = Acts::SurfaceArrayNavigationPolicy;
+    return Acts::NavigationPolicyFactory::make()
+        .add<Acts::TryAllNavigationPolicy>(
+            Acts::TryAllNavigationPolicy::Config{.sensitives = false})
+        .add<SrfArrayNavPol>(
+            SrfArrayNavPol::Config{.layerType = layerType, .bins = bins})
+        .asUniquePtr();
+  }
+};
+
+std::unique_ptr<const Acts::TrackingGeometry>
+Gen3GenericDetectorBuilder::buildTrackingGeometry(
+    const Acts::GeometryContext& gctx,
+    std::optional<std::filesystem::path> graphvizFile) {
+  using enum Acts::AxisDirection;
+  using namespace Acts::UnitLiterals;
+  using namespace Acts::Experimental;
+  using enum Acts::AxisBoundaryType;
+  using enum Acts::CylinderVolumeBounds::Face;
+  ACTS_INFO("GenericDetector construction in  Gen3 mode");
+
+  Blueprint::Config cfg;
+  cfg.envelope = Acts::ExtentEnvelope{{
+      .z = {20_mm, 20_mm},
+      .r = {0_mm, 20_mm},
+  }};
+  Blueprint root{cfg};
+
+  // @TODO: Add beampipe passive layer
+  // @TODO: Add Pixel
```

Comment:
```suggestion
```

Reply:
I suppose that's because it's done already, so yes, you can just remove it then.